docs/developer-guide.md

```diff
@@ -185,52 +185,197 @@ TestUtils.mock_postal_code()
+
+## Introduction
+
+Playwright is a powerful end-to-end testing library for web applications. It provides a simple API to automate browser actions, allowing developers to create robust and maintainable tests.
+
+## Getting Started
+
+To get started with Playwright, you'll need to install it as a dependency in your project:
+
 ```bash
 cd bc_obps && make run
+npm install @playwright/test
 ```
 
 2.0 Ensure the client app is running:
 
+## Writing Tests
+
+### Basic Test Structure
+
+Playwright tests follow a simple structure. Each test case is a JavaScript or TypeScript function with Playwright APIs for interacting with the browser.
+
+```javascript
+const { test } = require("@playwright/test");
+test("Example Test", async ({ page }) => {
+  // Your test logic goes here
+});
+```
+
+### Locators
+
+#### Prefer User-Facing Attributes
+
+When selecting elements, prefer using user-facing attributes over XPath or CSS selectors. This ensures that the tests are tied to the actual user experience and are less prone to breakage when the underlying structure changes.
+
+```javascript
+// Bad: Using XPath
+await page.click('//div[@class="my-button"]');
+// Good: Using User-Facing Attribute
+await page.click('[data-testid="my-button"]');
+```
+
+#### Code Generation
+
+To find locators, leverage Playwright's code generation feature. Use the following command to record user actions and generate test code:
+
+```bash
+cd client && npx playwright codegen http://localhost:3000
+```
+
+### Best Practices
+
+#### Test Isolation
+
+Ensure that each test is independent and does not rely on the state or outcome of other tests. This helps in maintaining a clean and predictable test suite.
+
+#### Use `expect` Assertions
+
+Use `expect` assertions to verify the expected behavior of the application. This makes your tests more readable and clearly defines the criteria for success.
+
+```javascript
+expect(await page.isVisible('[data-testid="success-message"]')).toBe(true);
+```
+
+#### Headless Mode for CI/CD
+
+When running tests in a Continuous Integration (CI) or Continuous Deployment (CD) pipeline, consider using headless mode to improve performance and resource consumption.
+
 Start client app from new terminal command:
 
+```javascript
+// Launch browser in headless mode
+const browser = await chromium.launch({ headless: true });
+```
+
+#### Playwright Testing Prerequisites
+
+- To run playwright end-to-end tests for the first time, you may need to run `yarn playwright install --with-deps` to install the browsers
+
+**Ensure you have the following configuration**
+
+1. From file `client/e2e/.env.local.example` create `client/e2e/.env.local` with values reflecting instructions in file `client/e2e/.env.local.example`
+2. Ensure the DB tables are configured with appropriate users and roles:
+
+```
+cd bc_obps && psql
+```
+
+```
+ \c registration
+```
+
+- Ensure the the erc.user table includes rows below **without** `bc-cas-dev-three`
```

Comment:
This seems a bit fragile/finicky--could we add something to the global that wipes the db and adds the desired fixtures so that the db is always in the appropriate state?

Reply:
BC so savvy, sooo savvy! 
:)